modules/e2ee/Context.js

```diff
@@ -150,14 +170,25 @@ export class Context {
                 }
             }
 
-            const frameTrailer = new Uint8Array(counterLength + 1);
+
+            let signatureLength = 0;
+
+            if (this._signatureKey) {
+                // TODO: the condition is more complex as we periodically want to
+                //   sign every n-th frame of every substream/ssrc.
```

Comment:
thinking of a `this._shouldSignFrame(encodedFrame.getMetadata())`
which looks at the metdadata.synchroniziationSource and does a time based decision